Backend/src/main/java/com/project/omega/controller/ClientController.java

```diff
@@ -0,0 +1,62 @@
+package com.project.omega.controller;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.project.omega.bean.dao.entity.Client;
+import com.project.omega.exceptions.ClientNotFoundException;
+import com.project.omega.exceptions.NoRecordsFoundException;
+import com.project.omega.repository.ClientRepository;
+import com.project.omega.service.interfaces.ClientService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+//import javax.lang.model.element.VariableElement;
+//import javax.xml.ws.Response;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/api/client")
+public class ClientController {
+
+    @Autowired
+    ClientService clientService;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+
+    @PostMapping(value = "/create", headers = "Accept=application/json")
+    public ResponseEntity<Client> createClient(@RequestBody Client client) {
+        Client newClient = clientService.createClient(client);
+        return new ResponseEntity(newClient, HttpStatus.CREATED);
+    }
+
+    @GetMapping(value = "/get")
+    public ResponseEntity<Client> GetAllClients() throws NoRecordsFoundException {
+        List<Client> clients = clientService.getAllClients();
+        return new ResponseEntity(clients, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/get/{id}")
+    public ResponseEntity getClient(@PathVariable(value = "id") Long id) throws ClientNotFoundException, NoRecordsFoundException {
+        Client client = clientService.getClientById(id);
+        return new ResponseEntity(client, HttpStatus.OK);
+
+    }
+
+    @DeleteMapping(value = "/delete/{id}")
+    public ResponseEntity deleteClient(@PathVariable(value = "id") Long id) throws ClientNotFoundException {
+        Optional<Client> client = clientService.deleteClientById(id);
+        return new ResponseEntity(client, HttpStatus.GONE);
```

Comment:
```suggestion
        return new ResponseEntity(client, HttpStatus.OK);
```
Please either use OK 200 or 204 ( No content ) when deleting gone indicates that the API recourse is gone